runtime/src/main/java/org/corfudb/protocols/wireprotocol/ICorfuPayload.java

```diff
@@ -124,6 +124,27 @@
         return builder.build();
     }
 
+    /** A TxResolutionInfo implementation.
```

Comment:
Do we really need a TXResolutionInfo type? It appears that this is just a way to abstract out the combination of the first read timestamp and the conflict map, which could easily just be part of the request as previous, with the timestamp and conflict map just set to null if not used.

Alternatively, you could have two message types: token_req and token_tx_req. Adding TXResolutionInfo as a type the serializer has to deal with seems like a bad idea.